### DISCUSSION TITLE
__call__ vs apply

### DISCUSSION THREAD

**Discussion by iislucas** [google/flax#808] [Q&A] [answered]
Hello! 

Newbie question... the flax basics shows a method in the nn.Module class called `__call__`, but when I look at the example here https://github.com/google/flax/blob/master/examples/seq2seq/train.py
The nn.Module's have `apply` methods, not `__call__` ones... what's the relationship between `apply` and `__call__`, if any? Looks like apply is flax, and __call__ is linen... ? a bit confused :-)

Thanks,

**Comment by iislucas** [comment]
Half answering myself, I see: https://github.com/google/flax/issues/571 so it looks like maybe this is just a API upgrade, with apply being the older style and __call__ being the new linen API.

**Comment by iislucas** [comment, marked as answer]
And I think the full answer for myself is to look at the `linen_examples` directory :) https://github.com/google/flax/blob/master/linen_examples/seq2seq/train.py

**Reply by avital** [reply]
Yup!
